src/Dex/Foreign/JIT.hs

```diff
@@ -123,9 +67,9 @@ dexFreeFunctionSignature sigPtr = do
   free =<< peekElemOff strPtr 2
   free sigPtr
 
-dexUnload :: Ptr JIT -> NativeFunctionAddr -> IO ()
-dexUnload jitPtr funcPtr = do
-  ForeignJIT{..} <- fromStablePtr jitPtr
-  addrTable <- readIORef addrTableRef
-  LLVM.JIT.unloadNativeModule $ nativeModule $ addrTable M.! funcPtr
-  modifyIORef addrTableRef $ M.delete funcPtr
+dexUnload :: Ptr Context -> ExportNativeFunctionAddr -> IO ()
+dexUnload ctxPtr funcPtr = do
+  Context _ _ ptrTabMVar <- fromStablePtr ctxPtr
+  addrTable <- takeMVar ptrTabMVar
+  putMVar ptrTabMVar $  M.delete funcPtr addrTable
```

Comment:
Since `insertIntoNativeFunctionTable` is a method of `Context`, perhaps make deletion from the table another method, instead of inlining it here? Otherwise it's inconsistent